[WP-1655][WP-1656][WP-1657] SDK documentation improvement (#9)
wazo-communication/developers.wazo.io: docs/sdk-librairies/plugins-js-sdk/plugins-apis.md

};
```

#### `removeModal`

> Will remove the shown modal in the application, the method must called from the `backgroundScript` file or within the modal HTML.

Reviewer: ```suggestion
Will remove the shown modal in the application, the method must be called from the `backgroundScript` file or within the modal HTML.
```

- `width` accepts valid CSS values, like `500px` or `80%`, used on Web/Desktop App (min: `300px`).
- `hideCloseButton` (default to false) determines if the close button should be displayed, or if the user should handle closing of the modal in the html content (through `app.removeModal()` method).

:::note
On Web/Desktop App, the banner will be integrated with other banners like incoming calls. Other banners will be displayed below.
:::

#### `closeBanner`

> We can hide a mobile banner fro the `backgroundScript` or withing the banner HTML page.

Reviewer: ```suggestion
We can hide a mobile banner from the `backgroundScript` or within the banner HTML page.
```

app.onParticipantLeaveRoom = (room, participant) => {};
```

## **E-UC Mobile** \{#mobile}

### Methods

#### `setMobileHeader`

Will change the header inside the app.

```js
app.setMobileHeader({ title?: string | null, callback?: Function | null });
```

- `title`: Changes the label displayed in the header. Set this value to `null` to set it back to the default value.
- `callback`: Handles the back button click. If a callback is registered, the app will forgo its normal behavior and run it. Setting the value to `null` reinstates the original behavior.

#### `setMobileShowBottomNav`

Define whether you want to show or hide the bottom navigation.

```js
app.setMobileShowBottomNav(show: boolean);
```

#### `startCall`

> Starting a call audio or video call

Reviewer: ```suggestion
Start a new audio or video call
```

```js
app.startCall({ targets , requestedModalities = ['audio'] });
```

Example:

```js
app.startCall({ targets: ['8008'], requestedModalities: ['video'] });
```

#### `playProgressSound`

Play the ringback sound when we make a call (🔁 loop)

```js
app.playProgressSound();
```

#### `playNewMessageSound`

Play the sound when we receive a text message.

```js
app.playNewMessageSound();
```

#### `playHangupSound`

Play the hangup sound

```js
app.playHangupSound();
```

#### `stopCurrentSound`

Stop the current sound. Sounds marked "🔁 loop" can be stopped only using this method.

```js
app.stopCurrentSound();
```

> #### `displayBanner`

Reviewer: This section could be moved next to `closeBanner`

We can display a mobile banner from the `backgroundScript` file.

```js
app.displayBanner({ url, height, width, hideCloseButton });
```

- If `url` is present, the modal will display an iframe with the content of the url.
- `height` accepts valid CSS values, like `500px` or `80%`.
- `width` accepts valid CSS values, like `500px` or `80%`, used on Web/Desktop App (min: `300px`).
- `hideCloseButton` (default to false) determines if the close button should be displayed, or if the user should handle closing of the modal in the html content (through `app.removeModal()` method).

:::note
On mobile, links with a `target="_blank"` attribute will be opened in the user's default browser.
:::

> #### `closeBanner`

Reviewer: Duplicate from l.417

Author: Yes, it's because we want available methods for both products `Web/Desktop` & `Mobile`  of the documentation.

:::note
On mobile, links with a `target="_blank"` attribute will be opened in the user's default browser.
:::

Example:

Displaying client data when receiving a call

```js
app.onCallIncoming = async call => {
  console.log('background onCallIncoming', call);
  const clientData = await fetchClientData(call.number); // Where `fetchClientData` is a method that return client information from an extension

  app.displayModal({
    title: `Incoming call for ${call.displayName}`,
    text: `Last call was: ${clientData.lastCall} for : ${clientData.lastCallSubject}`,
    height: '50%',
    width: '70%',
  });
};
```

#### `removeModal`

Will remove the shown modal in the application, the method must called from the `backgroundScript` file or within the modal HTML.

```js
app.removeModal();
```

#### `updateBadge`

Display a notification next to the navigation bar icon button. It uses MUI's [Badge](https://mui.com/material-ui/react-badge/) component to display a notification, generally an integer.

> When used in `backgroundScript` file, it requires `entityId` as set in a tab in the `staticTabs` section of `manifest.json`, and that tab must include "sidebarTab" in its context (`entityId` should automatically be set when called from the iframe).

Reviewer: ```suggestion
When used in `backgroundScript` file, it requires `entityId` to be set in a tab in the `staticTabs` section of `manifest.json`, and that tab must include "sidebarTab" in its context (`entityId` should automatically be set when called from the iframe).
```

```js
app.updateBadge({ badgeContent: number, entityId: string });
```

:::info
On Wazo Mobile, only the field `badgeContent: number` is used to display the badge.
:::

#### `sendMessageToBackground`

Sending a message from the iframe to background script. Receive message from event `onBackgroundMessage`

```js
// my-plugins.js (Sending message to background)
app.sendMessageToBackground({ value: 'ping' });

// background.js (Receiving a message from the iframe)
app.onBackgroundMessage = msg => {
  console.log('onBackgroundMessage', msg);

```

#### `sendMessageToIframe`

In the background script

```js
// background.js (Sending a message to the plugin's iframe)
app.sendMessageToIframe({ value: 'pong' });

// my-plugin.js (Receiving a message from the background script)
app.onIframeMessage = (msg) => {
  console.log('onIframeMessage', msg);
}
```

### Events

#### `onCallIncoming`

> Listener when a call for the current user is incoming. Useful to react on a call event from `backgroundScript` file.

Reviewer: ```suggestion
Listener to detect incoming calls for the current user. Useful to react on a call event from `backgroundScript` file.
```

```js
app.onCallIncoming = (call: Call) =>  {};
```

#### `onCallMade`

> Listener when a call is made by the user (eg: outgoing call). Useful to react on a call event from `backgroundScript` file.

Reviewer: ```suggestion
Listener to detect when a call is made by the user (eg: outgoing call). Useful to react on a call event from `backgroundScript` file.
```

```js
app.onCallMade = (call: Call) =>  {};
```

#### `onCallAnswered`

> Listener when an incoming call is answered by the current user. Useful to react on a call event from `backgroundScript` file.

Reviewer: ```suggestion
Listener to detect when an incoming call is answered by the current user. Useful to react on a call event from `backgroundScript` file.
```

```js
app.onCallAnswered = (call: Call) =>  {};
```

#### `onCallAccepted`

> Listener when a call is answered (by one or the other side). Useful to react on a call event from `backgroundScript` file.

Reviewer: ```suggestion
Listener  to detect answered calls (by the current user or the other side). Useful to react on a call event from `backgroundScript` file.
```

```js
app.onCallAccepted = (call: Call) =>  {};
```

#### `onCallHungUp`

> Listener when a call is hung up. Useful to react on a call event from `backgroundScript` file.

Reviewer: ```suggestion
Listener to detect hung up calls. Useful to react on a call event from `backgroundScript` file.
```

```js
app.onCallHungUp = (call: Call) =>  {};
```

#### `onWebsocketMessage`

Listener to Wazo Websocket message(s). Useful to react on a call event from `backgroundScript` file.

```js
app.onWebsocketMessage = (message: MessageEvent) =>  {};
```

## **E-UC Portal** {#portal}

### Methods

#### `changeToolbarDisplay`

Changes the status/visibility of the toolbar.

```js
app.changeToolbarDisplay(displayed: boolean);
```

### Events

#### `onConnectedToStack`

> Listener when the user is successfuly connected to a stack.

Reviewer: ```suggestion
Listener to detect when the user successfuly connects to a stack.
```
